### ISSUE TITLE
DOMException: Failed to set the 'selectionStart' property on 'HTMLInputElement'

### ISSUE THREAD

**Issue by sabernhardt** [WordPress/gutenberg#26983]
Reported by bradmarshall:
https://core.trac.wordpress.org/ticket/51577

-----

Actual behaviour:
dom.js is attempting to access the `selectionStart` property on input[type=email] in Gutenberg running WordPress 5.5.1, which triggers a DOM exception in the WP admin which breaks the gutenberg block containing the input.

Why?
This MDN article (https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange) states that only certain types of inputs and textareas impmlement selectionStart / selectionEnd but it looks as though numerous functions throughout the file attempt to use selectionStart/selectionEnd without first qualifying the type of input in question. Therefore, any input[type=email] elements in a block are crashing out.

Expected behaviour:
dom.js should not attempt to use selectionStart/selectionEnd on just any type of input - it should instead first check to make sure it's one of type=text, search, URL, tel or password. Gutenberg blocks containing input types other than these should not crash.

Version: 
WordPress 5.5.1

**Comment by talldan** [comment]
#26074 should fix this if we can resolve the feedback in the PR.

**Comment by kathrynwp** [comment]
Hi @talldan if there's any way to get that PR reviewed/merged, it would be great to be able to close this one out. Thanks!

**Comment by ndiego** [comment]
This issue was reviewed in today's Editor Bug Scrub. Upon investigation, it appears to have been resolved as part of https://github.com/WordPress/gutenberg/pull/40192 based on [this comment.](https://github.com/WordPress/gutenberg/pull/26074#issuecomment-1356812534) Therefore, I am going to close this issue but feel free to reopen it if issues persist. Thanks!
